### PR TITLE
[GISel] Improve MachineVerifier for G_SCMP/UCMP.

### PR DESCRIPTION
**Pull request by topperc**
Ensure destination type is at least 2 bits.
Remove unnecessary check that both sources are the same type. The verifier already handles this generically.

### CODE REVIEW

File: llvm/lib/CodeGen/MachineVerifier.cpp [hunk 1]
```diff
@@ -1627,18 +1626,18 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
       break;
     }
 
+    if (DstTy.getScalarSizeInBits() < 2) {
+      report("result type must be at least 2 bits wide", MI);
```
Comment (s-barannikov) [inline llvm/lib/CodeGen/MachineVerifier.cpp:1630]: It looks like most error messages start with a capital letter.
```suggestion
      report("Result type must be at least 2 bits wide", MI);
```

**Review by s-barannikov — APPROVED** [review]
LGTM

File: llvm/lib/CodeGen/MachineVerifier.cpp [hunk 2]
```diff
@@ -1627,18 +1626,18 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
...
+      break;
+    }
+
     if ((DstTy.isVector() != SrcTy.isVector()) ||
         (DstTy.isVector() &&
          DstTy.getElementCount() != SrcTy.getElementCount())) {
       report("Generic vector scmp/ucmp must preserve number of lanes", MI);
       break;
     }
```
Comment (s-barannikov) [inline llvm/lib/CodeGen/MachineVerifier.cpp:1639]: #120014 removes this check for G_ABDS/G_ABDU. Is it unnecessary or did another PR remove it unnecessarily?

Comment (s-barannikov) [reply llvm/lib/CodeGen/MachineVerifier.cpp:1639]: Don't mind me. G_ABDS/G_ABDU have the same type for result/operands, G_UCMP/G_SCMP don't and therefore require the check.

File: llvm/lib/CodeGen/MachineVerifier.cpp (hunk 1 above)
Comment (arsenm) [reply llvm/lib/CodeGen/MachineVerifier.cpp:1630]: The programming guide actually states to use lowercase error messages, but there are many violations

**Pull request merged**